### PR TITLE
Kazoo 3496

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

**Comment by fenollp** [comment]
Looks great to me! (also I need this)

File: applications/crossbar/src/modules/provisioner_v5.erl [hunk 1]
```diff
@@ -310,6 +309,82 @@ settings_datetime(JObj) ->
...
+            Type = wh_json:get_binary_value(<<"type">>, Value),
+            V = wh_json:get_binary_value(<<"value">>, Value),
+            FeatureKey = get_feature_key(Type, V, Brand, Family, AccountId),
+            maybe_add_feature_key(Key, FeatureKey, Acc)
+        end
+        ,wh_json:new()
+        ,FeatureKeys
+    ).
+
+-spec get_feature_key(ne_binary(), wh_json:object(), ne_binary(), ne_binary(), ne_binary()) -> wh_json:object().
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/provisioner_v5.erl]: Wrong spec. `Value` isn't a JSON object, is it? And `Brand` isn't ne_binary(). And `Family` isn't ne_binary()

Comment (jamesaimonetti) [inline applications/crossbar/src/modules/provisioner_v5.erl]: It also can return 'undefined' so should be `api_object()` for the return.

File: applications/crossbar/src/modules/provisioner_v5.erl [hunk 2]
```diff
@@ -310,6 +309,82 @@ settings_datetime(JObj) ->
...
+    end;
+get_feature_key(<<"parking">> = Type, Value, Brand, Family, _AccountId) ->
+    wh_json:from_list([
+        {<<"label">>, <<"Park ", Value/binary>>}
+        ,{<<"value">>, <<"*3", Value/binary>>}
+        ,{<<"type">>, get_feature_key_type(Type, Brand, Family)}
+    ]).
+
+-spec get_feature_key_type(ne_binary(), ne_binary(), ne_binary()) -> wh_json:object().
+get_feature_key_type(Type, Brand, Family) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/provisioner_v5.erl]: I think `Brand` and `Family` are JSON objects, no? And the return could be 'undefined'

File: applications/crossbar/src/modules/provisioner_v5.erl [hunk 3]
```diff
@@ -310,6 +309,82 @@ settings_datetime(JObj) ->
...
+    wh_json:get_first_defined([[Brand, Family, Type], [Brand, <<"_">>, Type]], ?FEATURE_KEYS).
+
+-spec get_user(ne_binary(), ne_binary()) -> wh_json:object().
+get_user(AccountId, UserId) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/provisioner_v5.erl]: `couch_mgr:open_doc/2` returns `{'ok', wh_json:object()} | {'error', _}`

File: applications/crossbar/src/modules/provisioner_v5.erl [hunk 4]
```diff
@@ -310,6 +309,82 @@ settings_datetime(JObj) ->
...
+    AccountDb = wh_util:format_account_id(AccountId, 'encoded'),
+    couch_mgr:open_doc(AccountDb, UserId).
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/provisioner_v5.erl]: Should probably use `open_cache_doc/2`

Comment (macpie) [inline applications/crossbar/src/modules/provisioner_v5.erl]: I want to keep it open_doc just to make sure data is accurate. Function is spawned anyway

**Pull request merged**